### ISSUE TITLE
일반 스트링에 대한 transform 적용 시 결과값 차이

### ISSUE THREAD

**Issue by namdori61** [SKTBrain/KoBERT#26]
안녕하세요, 먼저 모델 공유해주셔서 정말 감사합니다.

Pytorch 예시로 네이버 영화 리뷰 데이터셋에 대한 코드를 보면,

데이터를 txt 에서 `gluonnlp.data.TSVDataset` 로 로드하고 `gluonnlp.data.BERTSentenceTransform` 으로 `transform` 하게 되면

정상적으로 토크나이징이 진행되는데요.

`gluonnlp.data.TSVDataset`을 사용하지 않고, 일반적인 스트링에 `gluonnlp.data.BERTSentenceTransform` 을 적용하면 반환되는 값에 차이가 있습니다.

(일반적으로 `input_id` 의 유효한 토큰 및 `valid_lenth`의 개수가 낮아집니다.)

예를 들어, 학습 데이터에서 첫번째 인스턴스인 "아 더빙.. 진짜 짜증나네요 목소리" 에 대해서,

기존 예시 대로 적용하면 `valid_lenth`가 15이지만

`transform('아 더빙.. 진짜 짜증나네요 목소리')`로 바로 적용할 경우, `valid_lenth` 가 3이 나오면서 유효한 토큰 개수도 3개로 작게 나옵니다.

혹시 관련해서 해결책이 있을까요? 아니면 파일 로드 시에 반드시 gluonnlp 를 사용해야 하나요?

**Comment by haven-jeon** [comment]
간단하게 reproduce할 수 있는 코드 부탁드립니다.

**Comment by namdori61** [comment]
안녕하세요, 회신 감사합니다.

코드는 아래와 같습니다.

`
import numpy as np
import torch
from torch.utils.data import Dataset
import gluonnlp as nlp

from kobert.utils import get_tokenizer
from kobert.pytorch_kobert import get_pytorch_kobert_model

bertmodel, vocab = get_pytorch_kobert_model()
tokenizer = get_tokenizer()
tok = nlp.data.BERTSPTokenizer(tokenizer, vocab, lower=False)

transform = nlp.data.BERTSentenceTransform(tokenizer=tok,
                                           max_seq_length=140,
                                           pad=True,
                                           pair=False)

from_string = transform('아 더빙.. 진짜 짜증나네요 목소리')


!wget https://www.dropbox.com/s/374ftkec978br3d/ratings_train.txt?dl=1

dataset_train = nlp.data.TSVDataset("ratings_train.txt?dl=1", field_indices=[1,2], num_discard_samples=1)

class BERTDataset(Dataset):
    def __init__(self, dataset, sent_idx, label_idx, bert_tokenizer, max_len,
                 pad, pair):
        transform = nlp.data.BERTSentenceTransform(
            bert_tokenizer, max_seq_length=max_len, pad=pad, pair=pair)

        self.sentences = [transform([i[sent_idx]]) for i in dataset]
        self.labels = [np.int32(i[label_idx]) for i in dataset]

    def __getitem__(self, i):
        return (self.sentences[i] + (self.labels[i], ))

    def __len__(self):
        return (len(self.labels))

data_train = BERTDataset(dataset_train, 0, 1, tok, 140, True, False)

train_dataloader = torch.utils.data.DataLoader(data_train, batch_size=1, num_workers=5)

from_dataset = next(iter(train_dataloader))

print(from_string) # string에서 바로 transform 한 결과
print(from_dataset) # gluonnlp.data.TSVDataset 을 거쳐서 transform 한 결과
`

**Comment by haven-jeon** [comment]
코드를 돌려보진 못했는데, 보니.. 아래와 같이 하셔야 될거 같네요. 

```
from_string` = transform(['아 더빙.. 진짜 짜증나네요 목소리'])
```

**Comment by namdori61** [comment]
말씀해주신대로 리스트에 넣은 후에 transform하니 잘 변환됩니다!
바쁘실텐데 답변해주셔서 감사합니다:)
